### DISCUSSION TITLE
I flashed incorrectly, LED partial working, now it has stopped, in lsusb board is not visible

### DISCUSSION THREAD

**Discussion by GermanAizek** [bitcraze/discussions#1152] [Q&A]
I previously wrote about issue flash firmware in issues https://github.com/bitcraze/crazyradio-firmware/issues/65, but in fact I have Crazyradio PA board. Here are pictures my board.

![20240115_233448](https://github.com/bitcraze/discussions/assets/21138600/bebf548a-5128-4da2-a7a9-159d8b0e08ce)

![20240115_233413](https://github.com/bitcraze/discussions/assets/21138600/cd5bf540-cea7-486f-bc48-4c210f6e6e15)

```
ubuntu@ubuntu-NB:~/GIT/crazyradio-firmware/firmware$ make CRPA=1
Crazyradio PA build
ubuntu@ubuntu-NB:~/GIT/crazyradio-firmware/firmware$ python3 ../usbtools/launchBootloader.py 
Error!, cannot find the CrazyRadio USB dongle!
```

```
ubuntu@ubuntu-NB:~/GIT/crazyradio-firmware/firmware$ lsusb
Bus 004 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 003 Device 003: ID 09da:2403 A4Tech Co., Ltd. 2.4G Device
Bus 003 Device 002: ID 0408:a060 Quanta Computer, Inc. HD Webcam
Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 002 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 001 Device 003: ID 04ca:3016 Lite-On Technology Corp. 
Bus 001 Device 002: ID 03f0:0024 HP, Inc KU-0316 Keyboard
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
```

**Comment by GermanAizek** [comment]
I checked on Windows LED is red, on ubuntu 22.04 nothing.

**Reply by ataffanel** [reply]
What firmware are you running on your Crazyradio PA? The Crazyradio should appear with a VID of 1915, both in firmware and bootloader mode. Moreover, only the RED led lit up is not a behavior that is implemented in any of out firmware but that we have seen when flashing third party security-oriented firmware.

If you have flashed another firmware, you can try to see with its developer how to get back in bootloader mode.

In all case, you likely need to 'recover' your bootloader. This has to be done over SPI by programming the nrf51 flash. You can find a raspberrypi script in the firmware repos: https://github.com/bitcraze/crazyradio-firmware/tree/master/fix_bootloader. This can be used out of the box on a raspberry pi or could be ported to any 3.3V microcontroller or SBC. If you do port it to another platform we would be very happy to merge a pull request for it!

**Reply by GermanAizek** [reply]
@ataffanel,

> Moreover, only the RED led lit up is not a behavior that is implemented in any of out firmware but that we have seen when flashing third party security-oriented firmware.

U are completely right, I used this firmware:
https://github.com/BastilleResearch/nrf-research-firmware

> If you do port it to another platform we would be very happy to merge a pull request for it!

https://github.com/bitcraze/crazyradio-firmware/issues/65#issuecomment-1864634587
Earlier, I wrote to you how can use script for Orange Pi GPIO, but linux lsusb has not seen board, so its efficiency has not been proven.

**Reply by ataffanel** [reply]
Are you able to use the teensy/arduino solution as documented on the other project readme? https://github.com/BastilleResearch/nrf-research-firmware?tab=readme-ov-file#flash-over-spi-using-a-teensy This would allow you to reflash either the research firmware or the Crazyradio firmware (directly from the release: https://github.com/bitcraze/crazyradio-firmware/releases/tag/0.53)

**Reply by GermanAizek** [reply]
its difficult to get board from us, and it is also very expensive to buy, others cheaply arduino, stm32 or PIC, i have arduino nano (atmega 328p) and STM32F103, gpio pinout equivalent with Teensy?

**Reply by ataffanel** [reply]
The readme I linked to seems to imply that any Arduino would do. I suggest you write to the BastilleResearch project for more information about what flashing their firmware did to the Crazyradio PA and how to flash something else on it. We have  no knowledge about this firmware and its flashing procedure so it is impossible for me to know if my recovery script will even work at all (the recovery script just write an instruction to jump to the bootloader, if the bootloader has been erased, it will not work)

**Comment by GermanAizek** [comment]
![image](https://github.com/bitcraze/discussions/assets/21138600/79ba2c06-3565-4640-a761-139c0960caf3)


On Windows, different behavior board does detect as Unknown USB-device (device descriptor request failed) on Zadig

Installing process:

![image](https://github.com/bitcraze/discussions/assets/21138600/4b6e47ea-ae2a-4051-9c02-ffede0734cd4)


Result:

![image](https://github.com/bitcraze/discussions/assets/21138600/dfef80c7-4a35-4028-b198-7d3a955c0353)

**Comment by knmcguire** [comment]
Hi! I just reopened the issue again since it is indeed a crazyradio PA (version 1). https://github.com/bitcraze/crazyradio-firmware/issues/65

I'm pretty sure that Zadig won't help if it is not able to detect the Crazyradio PA.

Currently, we are very busy with finishing a deadline, so I'm marking your issue for a triage meeting that will be handled hopefully next week.

**Reply by GermanAizek** [reply]
ok i wait

**Comment by GermanAizek** [comment]
@knmcguire,
I have multimeter, I can measure resistance, voltage and capacity smd components, I can also replace them.
If problem is not software-based.
